azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/containerapp/ContainerAppDraft.java

```diff
@@ -275,6 +278,73 @@ private static void tarSourceIfNeeded(final BuildImageConfig buildConfig) {
+        final ContainerRegistry registry = getOrCreateRegistry(imageConfig);
+        tarSourceIfNeeded(buildConfig);
+        final RegistryTaskRun run = registry.buildImage(imageConfig.getAcrImageNameWithTag(), buildConfig.getSource(), "./Dockerfile", overridingArguments);
+        if (Objects.isNull(run)) {
+            throw new AzureToolkitRuntimeException("ACR is not ready, Failed to build image through ACR.");
+        }
+        return registry.waitForImageBuilding(run);
+    }
+
+    private static void deleteTempFolder(Path tempFolder) {
+        if (Objects.isNull(tempFolder)) {
+            return;
+        }
+        try {
+            FileUtils.deleteDirectory(tempFolder.toFile());
+        } catch (IOException e) {
+            throw new AzureToolkitRuntimeException("Failed to delete temporary directory: " + tempFolder, e);
+        }
+    }
+
+    private static void generateDockerfile(final BuildImageConfig buildConfig, String templatePath) {
+        Path destination = buildConfig.source.resolve("Dockerfile");
+        // Path to the Dockerfile inside the resources/template folder
+        // Load the Dockerfile from the resources
+        InputStream inputStream = ContainerAppDraft.class.getClassLoader().getResourceAsStream(templatePath);
+        if (inputStream == null) {
+            throw new AzureToolkitRuntimeException("Template dockerfile not found in the resources: " + templatePath);
+        }
+        // Copy the Dockerfile to the destination path
+        try {
+            Files.copy(inputStream, destination, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new AzureToolkitRuntimeException("Failed to copy Dockerfile to the destination path: " + destination, e);
+        }
+        AzureMessager.getMessager().info("Dockerfile generated successfully to: " + destination);
+    }
+
+    private static Path generateTempFolder(final BuildImageConfig buildConfig) {
+        // Create a temporary directory and handle resources
+        Path tempDir;
+        try {
+            // Step 1: Create a temporary directory
+            tempDir = Files.createTempDirectory(String.format("aca-maven-plugin-%s", Utils.getTimestamp()));
+            AzureMessager.getMessager().info("Temporary directory created: " + tempDir);
+
+            // Step 2: Copy Jar to the temporary directory
+            Path sourceFile = buildConfig.getSource();  // replace with your file path
+            Path targetFile = tempDir.resolve("app.jar");
+            Files.copy(sourceFile, targetFile, StandardCopyOption.REPLACE_EXISTING);
+            AzureMessager.getMessager().info("File copied to temporary directory: " + targetFile);
+
+            buildConfig.setSource(tempDir);
+
+            generateDockerfile(buildConfig, artifactDockerFilePath);
+
+        } catch (IOException e) {
+            throw new AzureToolkitRuntimeException("Failed to create temporary directory and copy artifact", e);
+        }
```

Comment:
We should have a finally step here to optionally clean up the tempDir, right?

Reply:
The tempDir will be deleted after being uploaded to ACR. Will add cleanup logic in catch part.